types/simple-oauth2/index.d.ts

```diff
@@ -1,58 +1,121 @@
         id: string;
-        /** Service registered client secret. Required. */
+        /** Service registered client secret. When required by the spec this value will be automatically encoded. Required. */
         secret: string;
-        /** Parameter name used to send the client secret. Default to client_secret. */
+        /** Parameter name used to send the client secret. Defaults to client_secret. */
         secretParamName?: string | undefined;
-        /** Parameter name used to send the client id. Default to client_id. */
+        /** Parameter name used to send the client id. Defaults to client_id. */
         idParamName?: ClientIdName | undefined;
     };
     auth: {
-        /** String used to set the host to request the tokens to. Required. */
+        /** Base URL used to obtain access tokens. Required */
         tokenHost: string;
-        /** String path to request an access token. Default to /oauth/token. */
+        /**
+         * URL path to obtain access tokens. Defaults to /oauth/token.
+         *
+         * Note: URL paths are relatively resolved to their corresponding host property using the Node WHATWG URL resolution algorithm
+         */
         tokenPath?: string | undefined;
-        /** String path to revoke an access token. Default to /oauth/revoke. */
+        /**
+         * URL path to refresh access tokens. Defaults to auth.tokenPath
+         *
+         * Note: URL paths are relatively resolved to their corresponding host property using the Node WHATWG URL resolution algorithm
+         */
+        refreshPath?: string | undefined;
+        /**
+         * URL path to revoke access tokens. Defaults to /oauth/revoke
+         *
+         * Note: URL paths are relatively resolved to their corresponding host property using the Node WHATWG URL resolution algorithm
+         */
         revokePath?: string | undefined;
-        /** String used to set the host to request an "authorization code". Default to the value set on auth.tokenHost. */
+        /** Base URL used to request an authorization code. Only valid for AuthorizationCode. Defaults to auth.tokenHost */
         authorizeHost?: string | undefined;
-        /** String path to request an authorization code. Default to /oauth/authorize. */
+        /**
+         * URL path to request an authorization code. Only valid for AuthorizationCode. Defaults to /oauth/authorize
+         *
+         * Note: URL paths are relatively resolved to their corresponding host property using the Node WHATWG URL resolution algorithm
+         */
         authorizePath?: string | undefined;
     };
     /**
      * Used to set global options to the internal http library (wreck).
      * All options except baseUrl are allowed
      * Defaults to header.Accept = "application/json"
      */
-    http?: {} | undefined;
+    http?: {
+        baseUrl?: undefined;
```

Comment:
shouldn't this be `string`?
```suggestion
        baseUrl?: string;
```

Reply:
@namdien177 As far as I understand, only undefined can be passed here.

Proofs:
https://github.com/lelylan/simple-oauth2/blob/master/lib/config.js#L67
![image](https://user-images.githubusercontent.com/5254026/216122019-8a2b370e-173c-443f-9cc7-dbe12f015014.png)